pkg/build/api/v1/conversion.go

```diff
@@ -114,6 +114,32 @@ func convert_v1_BuildTriggerPolicy_To_api_BuildTriggerPolicy(in *BuildTriggerPol
 	return nil
 }
 
+func convert_v1_BuildConfigSpec_To_api_BuildConfigSpec(in *BuildConfigSpec, out *newer.BuildConfigSpec, s conversion.Scope) error {
+	if err := s.DefaultConvert(in, out, conversion.DestFromSource); err != nil {
+		return err
+	}
+	// Remove unrecognized trigger types
+	triggers := []newer.BuildTriggerPolicy{}
+	for _, t := range out.Triggers {
+		switch t.Type {
+		case newer.ImageChangeBuildTriggerType,
+			newer.GenericWebHookBuildTriggerType,
+			newer.GitHubWebHookBuildTriggerType,
+			newer.ConfigChangeBuildTriggerType:
+			triggers = append(triggers, t)
```

Comment:
add a comment in api package by the constants that we have to update this whenever we add a new one?